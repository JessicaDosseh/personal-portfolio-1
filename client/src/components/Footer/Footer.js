import React from "react";
import styled, { keyframes } from "styled-components";
import { Link } from "react-router-dom";
import { css } from "styled-components";
import githubLogo from "./../../images/github.svg";
import githubLogoWhite from "./../../images/github-white.svg";
import linkedIn from "./../../images/linkedin.svg";
import linkedInWhite from "./../../images/linkedin-white.svg";
import medium from "./../../images/medium.svg";
import mediumWhite from "./../../images/medium-white.svg";
import twitter from "./../../images/twitter.svg";
import twitterWhite from "./../../images/twitter-white.svg";
import instagram from "./../../images/instagram.svg";
import instagramWhite from "./../../images/instagram-white.svg";
import mail from "./../../images/envelope.svg";
import mailWhite from "./../../images/envelope-white.svg";
import PropTypes from "prop-types";

const hoverLink = keyframes`
  0% {
    color: black;
  }
  100% {
    color: #AD91ED;
  }
`;

const hoverLinkDarkmode = keyframes`
  0% {
    color: white;
  }
  100% {
    color: #6DBCF5;
  }
`;

const contactHover = keyframes`
  0% {
    color: #5A22DB;
  }
  100% {
    color: #AD91ED;
  }
`;

const contactHoverDarkmode = keyframes`
  0% {
    color: #1a97f0;
  }
  100% {
    color: #6DBCF5;
  }
`;

const FooterContainer = styled.div`
  width: 100%;
  max-width: 1200px;
  margin: 70px auto 0;
  display: flex;
  flex-direction: column;

  ${props =>
    props.darkmode &&
    css`
      color: white;
    `}

  div:first-child {
    @media (min-width: 800px) {
      display: flex;
    }

    p {
      margin-top: 20px;
    }

    a {
      color: #6936de;
      display: block;
      margin: 30px 0 10px;
      font-size: 3rem;
      line-height: 1.8;

      @media (min-width: 800px) {
        margin: 20px 10px 0;
      }

      ${props =>
        props.darkmode &&
        css`
          color: #1a97f0;
        `}

      &:hover {
        ${props =>
          props.darkmode
            ? css`
                animation: ${contactHoverDarkmode} 0.5s forwards;
              `
            : css`
                animation: ${contactHover} 0.5s forwards;
              `}
      }

      &:active {
        ${props =>
          props.darkmode
            ? css`
                color: #b6465f;
              `
            : css`
                color: #ad91ed;
              `}
      }
    }
  }
`;

const DesktopLinks = styled.div`
  display: none;

  @media (min-width: 800px) {
    flex-direction: row;
    display: flex;
    flex-direction: row;

    a {
      margin-top: 30px;
      font-size: 1.6rem;

      @media (min-width: 800px) {
        margin: 50px 3% 120px 0;
      }

      ${props =>
        props.darkmode &&
        css`
          color: white;
        `}

      &:hover {
        ${props =>
          props.darkmode
            ? css`
                animation: ${hoverLinkDarkmode} 0.2s forwards;
              `
            : css`
                animation: ${hoverLink} 0.2s forwards;
              `}
      }
    }

    a:last-child {
      margin-bottom: 80px;
    }
  }
`;

const MobileLinks = styled.div`
  width: 60px;
  margin-bottom: 10px;

  a {
    display: inline-block;
    margin: 10px 0;

    &:hover {
      opacity: 0.8;
    }
  }

  a:last-child {
    margin-top: 5px;
  }

  @media (min-width: 800px) {
    display: none;
  }
`;

const Copyright = styled.div`
  max-width: 1200px;
  width: 100%;
  margin: 0 auto;
  display: flex;
  justify-content: center;

  h4 {
    display: block;
    width: 100%;
    text-align: center;
    font-size: 1.6rem;
    line-height: 1.8;
    margin-bottom: 80px;
    color: black;

    @media (min-width: 500px) {
      font-size: 1.6rem;
    }

    ${props =>
      props.darkmode &&
      css`
        color: white;
      `}
  }
`;

const Footer = props => {
  return (
    <>
      <FooterContainer darkmode={props.darkmode}>
        <div>
          <p>Looking to start a project?</p>
          <Link to="/contact">Get in contact.</Link>
        </div>
        <DesktopLinks darkmode={props.darkmode}>
          <a href="https://github.com/nwthomas">GitHub</a>
          <a href="https://www.linkedin.com/in/nwthomas-profile/">LinkedIn</a>
          <a href="https://medium.com/@nwthomas">Medium</a>
          <a href="https://twitter.com/nwthomas_">Twitter</a>
          <a href="https://www.instagram.com/nwthomas/">Instagram</a>
          <Link to="/contact">Email</Link>
        </DesktopLinks>
        <MobileLinks>
          <a href="https://github.com/nwthomas">
            <img
              src={props.darkmode ? githubLogoWhite : githubLogo}
              alt="Github logo"
            />
          </a>
          <a href="https://www.linkedin.com/in/nwthomas-profile/">
            <img
              src={props.darkmode ? linkedInWhite : linkedIn}
              alt="LinkedIn logo"
            />
          </a>
          <a href="https://medium.com/@nwthomas">
            <img
              src={props.darkmode ? mediumWhite : medium}
              alt="Medium logo"
            />
          </a>
          <a href="https://twitter.com/nwthomas_">
            <img
              src={props.darkmode ? twitterWhite : twitter}
              alt="Twitter logo"
            />
          </a>
          <a href="https://www.instagram.com/nwthomas/">
            <img
              src={props.darkmode ? instagramWhite : instagram}
              alt="Instagram logo"
            />
          </a>
          <Link to="/contact">
            <img src={props.darkmode ? mailWhite : mail} alt="Mail icon" />
          </Link>
        </MobileLinks>
      </FooterContainer>
      <Copyright darkmode={props.darkmode}>
        <h4>Copyright Â© 2019 Nathan Thomas</h4>
      </Copyright>
    </>
  );
};

Footer.propTypes = {
  darkmode: PropTypes.bool.isRequired
};

export default Footer;
