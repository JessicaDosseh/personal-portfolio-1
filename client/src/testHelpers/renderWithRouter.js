import React from "react";
import { Router } from "react-router-dom";
import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
