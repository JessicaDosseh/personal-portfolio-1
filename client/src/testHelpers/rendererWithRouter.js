import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { globalReducer, globalContext } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={globalContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
