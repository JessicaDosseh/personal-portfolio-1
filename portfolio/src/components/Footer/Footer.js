import React from "react";
import styled, { keyframes } from "styled-components";
import { Link } from "react-router-dom";
import { css } from "styled-components";

const hoverLink = keyframes`
  0% {
    color: black;
  }
  100% {
    color: #AD91ED;
  }
`;

const hoverLinkDark = keyframes`
  0% {
    color: white;
  }
  100% {
    color: #c3677c;
  }
`;

const FooterContainer = styled.div`
  width: 90%;
  margin: 40px 5% 0;
  display: flex;
  flex-direction: column;

  ${props =>
    props.darkmode &&
    css`
      color: white;
    `}

  p {
    margin-top: 20px;
  }

  p:nth-child(even) {
    color: #6936de;
    display: block;
    margin: 60px 0 10px;

    ${props =>
      props.darkmode &&
      css`
        color: #b6465f;
      `}
  }

  a {
    margin-top: 30px;
    font-size: 1.6rem;

    ${props =>
      props.darkmode &&
      css`
        color: white;
      `}

    &:hover {
      ${props =>
        props.darkmode
          ? css`
              animation: ${hoverLinkDark} 0.5s forwards;
            `
          : css`
              animation: ${hoverLink} 0.5s forwards;
            `}
    }
  }

  a:last-child {
    margin-bottom: 80px;
  }
`;

const Footer = props => {
  return (
    <FooterContainer darkmode={props.darkmode}>
      <p>Looking to start a project?</p>
      <p>Get in contact.</p>
      <a href="https://github.com/nwthomas">GitHub</a>
      <a href="https://www.linkedin.com/in/nathan-thomas-644b3339/">LinkedIn</a>
      <a href="https://medium.com/@nwthomas">Medium</a>
      <a href="https://twitter.com/_nwthomas">Twitter</a>
      <a href="https://www.instagram.com/nwthomas/">Instagram</a>
      <Link to="/contact">Email</Link>
    </FooterContainer>
  );
};

export default Footer;
