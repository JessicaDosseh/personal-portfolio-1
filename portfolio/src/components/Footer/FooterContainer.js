import React from "react";
import styled, { keyframes } from "styled-components";
import { Link } from "react-router-dom";
import { css } from "styled-components";

const hoverLink = keyframes`
  0% {
    color: black;
  }
  100% {
    color: #AD91ED;
  }
`;

const FooterContainerDiv = styled.div`
  width: 90%;
  margin: 40px 5% 0;
  display: flex;
  flex-direction: column;

  ${props =>
    props.darkmode &&
    css`
      color: white;
    `}

  p {
    margin-top: 20px;
  }

  p:nth-child(even) {
    color: #6936de;
    display: block;
    margin: 60px 0 10px;

    ${props =>
      props.darkmode &&
      css`
        color: #b6465f;
      `}
  }

  a {
    margin-top: 30px;
    font-size: 1.6rem;

    ${props =>
      props.darkmode &&
      css`
        color: white;
      `}

    &:hover {
      animation: ${hoverLink} 0.35s forwards;
    }
  }

  a:last-child {
    margin-bottom: 80px;
  }
`;

const FooterContainer = props => {
  return (
    <FooterContainerDiv darkmode={props.darkmode}>
      <p>Looking to start a project?</p>
      <p>Get in contact.</p>
      <a href="https://github.com/nwthomas">GitHub</a>
      <a href="https://www.linkedin.com/in/nathan-thomas-644b3339/">LinkedIn</a>
      <a href="https://medium.com/@nwthomas">Medium</a>
      <a href="https://twitter.com/_nwthomas">Twitter</a>
      <a href="https://www.instagram.com/nwthomas/">Instagram</a>
      <Link to="/contact">Email</Link>
    </FooterContainerDiv>
  );
};

export default FooterContainer;
