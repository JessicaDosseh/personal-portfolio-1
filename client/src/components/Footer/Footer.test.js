import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import Footer from "./Footer";

describe("<Footer />", () => {
  it("should render the Footer component without crashing", () => {
    const helpers = renderWithRouter(<Footer darkmode={false} />);
  });

  it("should match the snapshot of Footer", () => {
    const tree = rendererWithRouter(<Footer darkmode={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render all footer links to the screen", () => {
    const { getByText } = renderWithRouter(<Footer darkmode={false} />);
    expect(getByText(/GitHub/i)).toBeDefined();
    expect(getByText(/LinkedIn/i)).toBeDefined();
    expect(getByText(/Medium/i)).toBeDefined();
    expect(getByText(/Twitter/i)).toBeDefined();
    expect(getByText(/Instagram/i)).toBeDefined();
    expect(getByText(/Email/i)).toBeDefined();
  });

  it("should render the copyright section to the screen", () => {
    const { getByText } = renderWithRouter(<Footer darkmode={false} />);
    expect(getByText(/Copyright Â© 2019 Nathan Thomas/i)).toBeDefined();
  });
});
