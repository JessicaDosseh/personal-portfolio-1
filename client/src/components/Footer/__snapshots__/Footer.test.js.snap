// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match the snapshot of Footer 1`] = `
Array [
  <div
    className="sc-AykKC ZICuv"
  >
    <div>
      <p>
        Looking to start a project?
      </p>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Get in contact.
      </a>
    </div>
    <div
      className="sc-AykKD fMrFOE"
    >
      <a
        href="https://github.com/nwthomas"
      >
        GitHub
      </a>
      <a
        href="https://www.linkedin.com/in/nwthomas-profile/"
      >
        LinkedIn
      </a>
      <a
        href="https://medium.com/@nwthomas"
      >
        Medium
      </a>
      <a
        href="https://twitter.com/nwthomas_"
      >
        Twitter
      </a>
      <a
        href="https://www.instagram.com/nwthomas/"
      >
        Instagram
      </a>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Email
      </a>
    </div>
  </div>,
  <div
    className="sc-AykKE gHKFZd"
  >
    <h4>
      Copyright Â© 2019 Nathan Thomas
    </h4>
  </div>,
]
`;
